src/PockyBot.NET/Services/Pegs/PegGiver.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Threading.Tasks;
+using GlobalX.ChatBots.Core;
+using GlobalX.ChatBots.Core.Messages;
+using PockyBot.NET.Persistence.Models;
+using PockyBot.NET.Persistence.Repositories;
+
+namespace PockyBot.NET.Services.Pegs
+{
+    internal class PegGiver : IPegGiver
+    {
+        private readonly IPegRepository _pegRepository;
+        private readonly IChatHelper _chatHelper;
+
+        public PegGiver(IPegRepository pegRepository, IChatHelper chatHelper)
+        {
+            _pegRepository = pegRepository;
+            _chatHelper = chatHelper;
+        }
+
+        public async Task<Message> GivePeg(string comment, PockyUser sender, PockyUser receiver, int numPegsGiven)
+        {
+            var peg = new Peg
+            {
+                Sender = sender.UserId,
+                Receiver = receiver.UserId,
+                Comment = comment
+            };
+            _pegRepository.CreatePeg(peg);
+
+            await PmSender(sender, receiver, numPegsGiven);
+            await PmReceiver(comment, sender, receiver);
+
+            return null;
```

Comment:
is this supposed to return something?

Reply:
Nope. Ideally it should probably return an error message if it fails, but on the successful case it should be silent

Reply:
Error messages are better passed through as actually throwing and catching errors. This Task<Message> can be changed into a Task instead if you weren't expecting a return value